File: Modules/OpenSource/src/main/java/com/example/william/my/open/activity/LiveEventBusActivity.java
package com.example.william.my.open.activity;

import androidx.annotation.Nullable;
import androidx.lifecycle.Observer;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.example.william.my.module.activity.ResponseActivity;
import com.example.william.my.module.router.ARouterPath;
import com.jeremyliao.liveeventbus.LiveEventBus;

/**
 * 消息TAG为一个String类型的消息名字
 * https://github.com/JeremyLiao/LiveEventBus
 */
@Route(path = ARouterPath.OpenSource.OpenSource_LiveEventBus)
public class LiveEventBusActivity extends ResponseActivity {

    @Override
    public void initView() {
        super.initView();

        LiveEventBus
                .get("some_key", String.class)
                .observe(this, new Observer<String>() {
                    @Override
                    public void onChanged(@Nullable String s) {
                        showResponse(s);
                    }
                });
    }

    @Override
    public void setOnClick() {
        super.setOnClick();
        LiveEventBus
                .get("some_key")
                //.post("LiveEventBus")
                .postDelay("LiveEventBus", 1000);
    }
}

File: modules/module_lib/src/main/java/com/example/william/my/module/libraries/activity/LiveEventBusActivity.kt
package com.example.william.my.module.libraries.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.alibaba.android.arouter.facade.annotation.Route
import com.example.william.my.core.eventbus.livedata.LiveEventBus
import com.example.william.my.module.libraries.databinding.LibActivityEventBusBinding
import com.example.william.my.module.libraries.event.GlobalEvent
import com.example.william.my.module.libraries.event.StickyEvent
import com.example.william.my.module.router.ARouterPath

@Route(path = ARouterPath.Lib.Lib_LiveEventBus)
class LiveEventBusActivity : AppCompatActivity() {

    private lateinit var mBinding: LibActivityEventBusBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        mBinding = LibActivityEventBusBinding.inflate(layoutInflater);
        setContentView(mBinding.root)

        postStickyEvent()

        observeEvent()

        postEvent()
    }

    private fun postStickyEvent() {
        LiveEventBus.postEvent(this, StickyEvent("send StickyEvent by Activity"))
    }

    private fun observeEvent() {
        LiveEventBus.observeEvent<GlobalEvent>(this) {
            mBinding.global.text = it.message
        }
        LiveEventBus.observeEvent<StickyEvent>(this, isSticky = true) {
            mBinding.sticky.text = it.message
        }
    }

    private fun postEvent() {
        LiveEventBus.postEvent(this, GlobalEvent("send GlobalEvent by Activity"))
    }
}